packages/-ember-data/tests/integration/identifiers/configuration-test.ts

```diff
@@ -507,7 +508,7 @@ module('Integration | Identifiers - configuration', function (hooks) {
     setIdentifierForgetMethod((identifier) => {
       forgetMethodCalls++;
       let expectedIdentifier = expectedIdentifiers.shift();
-      assert.ok(expectedIdentifier === identifier, `We forgot the expected identifier ${expectedIdentifier}`);
+      assert.strictEqual(expectedIdentifier, identifier, `We forgot the expected identifier ${expectedIdentifier}`);
```

Comment:
This improvement has uncovered a type mismatch here.

```
$ tsc -p tsconfig.json --noEmit --pretty false
Error: packages/-ember-data/tests/integration/identifiers/configuration-test.ts(511,4[6](https://github.com/emberjs/data/runs/5158545581?check_suite_focus=true#step:7:6)): error TS2345: Argument of type 'StableIdentifier | StableRecordIdentifier' is not assignable to parameter of type 'StableRecordIdentifier | undefined'.
  Type 'StableIdentifier' is not assignable to type 'StableRecordIdentifier | undefined'.
    Type 'StableIdentifier' is missing the following properties from type 'StableNewRecordIdentifier': id, type
```

Reply:
Given this works in the previous test when expectedIdentifier is not defined the easy fix is to change the type of the array. This seems like a bug in qunit's types to me though, the comparisons should be comparing `unknown` and not attempting to do a type comparison.

What think you @chriskrycho ?

Reply:
I think the issue is that `expectedIdentifiers.shift()` could theoretically return `undefined` if the array is empty. So I added an `if` statement to avoid this assertion if the array is somehow empty.